### ISSUE TITLE
Branch  no longer differs from base branch

### ISSUE THREAD

**Issue by k0da** [peter-evans/create-pull-request#902]
### Subject of the issue

During release we generate offline changelog and want to open PR with changes made for manual review. Action runs on new tag event end checks out specific tag, changelog generated and commited during action run locally, then we want to open a PR with this commit against master. Despite job status is green, no PR has been created

### Steps to reproduce

See [affected job](https://github.com/k8gb-io/k8gb/blob/master/.github/workflows/changelog_pr.yaml#L8) and latest [run](https://github.com/k8gb-io/k8gb/runs/2819829287?check_suite_focus=true#step:6:82)

**Comment by peter-evans** [comment]
Hi @k0da 

Thank you for reporting this. I think it might be a bug and I'm currently investigating.

I think this bug has surfaced because of the slightly unusual conditions in this use case and the way that you are using the action. You don't need to commit the file in a separate step because the action will do that for you. Please change your workflow as follows.

```diff
      - uses: heinrichreimer/github-changelog-generator-action@v2.2
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          project: k8gb
          pullRequests: true
          author: true
          issues: true
          issuesWoLabels: true
          prWoLabels: true
          compareLink: true
          filterByMilestone: true
          unreleased: true
-     - name: Commit changelog
-       run: |
-         git config user.name 'Github Action'
-         git config user.email 'action@users.noreply.github.com'
-         git add CHANGELOG.md
-         git commit -m "Update Offline Changelog"
      - name: Create Pull Request
        uses: peter-evans/create-pull-request@v3
        with:
          title: "Update Offline Changelog"
+         commit-message: Update Offline Changelog
+         author: Github Action <action@users.noreply.github.com>
          branch: offline_changelog
          delete-branch: true
          base: master
          token: ${{ secrets.GITHUB_TOKEN }}
```
I think making the above change will solve the problem.

**Comment by peter-evans** [comment]
I've just realised that I already knew about this limitation because there is note in the documentation about this [here](https://github.com/peter-evans/create-pull-request#controlling-commits).

> Note that the repository must be checked out on a branch with a remote, it won't work for events which checkout a commit.

I'll try and explain what is happening. The action does support creating your own commits (like the additional step you have), but the action can only find those commits if they were committed to a branch. Your workflow is triggered by `release` events. In those events the `actions/checkout` action by default checks out the release tag. This is checking out the repository on a commit, not a branch, and it leaves the repository in a "detached HEAD" state. If you commit in a detached HEAD state, the HEAD ref moves to the latest commit. This prevents the action being able to track the commits you made.

So to summarise. This is a limitation of the feature to control your own commits. It can't be used when the repository is checked out on a commit (detached HEAD).

In your case the solution is very simple. Just make the changes I proposed to remove the additional step and let the action commit for you.

**Comment by k0da** [comment]
@peter-evans Thanks, I realized that yesterday by playing with a fork a little bit and figured out the same. https://github.com/k0da/k8gb/blob/master/.github/workflows/changelog_pr.yaml 

Maybe such case should be explained in more details? Anyway I think issue can be closed now.

**Comment by peter-evans** [comment]
Unfortunately, I don't think there is a good way for the action to detect this case and warn the user. Hopefully other users that run into this limitation will see the note in the documentation and/or find this issue. I don't think this will be very common anyway.
